addons/account/models/account_move.py

```diff
@@ -1793,6 +1794,14 @@ def _build_credit_warning_message(self, record, updated_credit):
         msg += ': %s' % formatLang(self.env, updated_credit, currency_obj=record.company_id.currency_id)
         return msg
 
+    @api.depends('line_ids')
+    def _compute_invoice_origin_count(self):
+        for move in self:
+            if move.move_type == 'in_invoice' and hasattr(move.line_ids, 'purchase_line_id'):
+                move.invoice_origin_count = len(move.line_ids.purchase_line_id.order_id)
+            elif move.move_type == 'out_invoice' and hasattr(move.line_ids, 'sale_line_ids'):
+                move.invoice_origin_count = len(move.line_ids.sale_line_ids.order_id)
```

Comment:
I think the dependency should be inverted, they should be added in purchase and sale modules not in account

Reply:
I have changed it - could please take a look again? :slightly_smiling_face: